pkg/bufferdecoder/eventsreader.go

```diff
@@ -373,21 +372,35 @@ func readStringFromBuff(ebpfMsgDecoder *EbpfDecoder) (string, error) {
 func readStringVarFromBuff(decoder *EbpfDecoder, max int) (string, error) {
 	var err error
 	var char int8
-	res := make([]byte, max)
+	var res []byte
+
+	switch {
+	case max > 0:
+		res = make([]byte, 0, max)
+	case max == 0:
+		res = make([]byte, 0)
```

Comment:
This change was based on the comment `max length can be passed as `max` to optimize memory allocation, otherwise pass 0`, but the logic below does the opposite when max is set as 0. The "Zero max length" unit test is the proof of it.

Then I ask what should be the expected behaviour when max is set as 0.

Reply:
Removed validation.